index.html

```diff
@@ -2859,8 +2859,8 @@ <h3>Production</h3>
         <p>
 The <a>DID document</a> MUST be serialized according to the
 <a href="#production">production rules for JSON</a>, with one additional
-requirement: The <a>DID document</a> MUST include the <code>@context</code>
-entry.
+requirement: The serialized <a>DID document</a> MUST include the
```

Comment:
This doesn't say where `@context` comes from. I don't think that's a big deal, JSON-LD folks will know to inject it if it doesn't exist in the ADM. However, there are cases where if you start with JSON-only AND don't include a `@context` AND want to be able to serialize to JSON-LD, it is possible to create a situation where you cannot serialize to JSON-LD (because you cannot know what the original intent of all of the terms were if `@context` was removed). Consider the following:

1. Assume three terms in two different contexts
2. termA in contextA, and termB in contextB
3. contextA and contextB each contain termC
4. contextA establishes termC to mean 'source'
5. contextB establishes termC to mean 'destination'
6. termA and termB are registered in DID Spec Registries
7. termC is not registered in DID Spec Registries
8. You have a DID Document that contains termA, termB, and termC

At this point, it's impossible to deterministically construct the `@context` for the JSON-LD representation using only the knowledge above. It is, therefore, impossible to use the DID Spec Registries to recreate `@context` for all valid DID Documents.

Folks in the WG seem to continue to be confused by this, thinking that the DID Spec registries can be used to recreate the `@context` field.

Reply:
I think in 99% of the DID documents we see today, adding the `@context` during JSON-LD production is trivial. One of the reasons why the DID Spec Registries were created was to help with constructing the `@context` if you don't have it (e.g. if you have an `ethereumAddress` property in the DID document). I do agree that this is not always possible if the DID document uses unregistered extensions.

Reply:
more importantly vanilla json does not care about the registries.... so you can very easily do this:

```ts
const didDocument = consume(json);
didDocument.foobarbaz = 123;
didDocument.produce(json-ld);
```

One of 2 things should happen...

1.    an error should be thrown `@context` is not defined.
2.  `didDocument.produce(json-ld, { `@context`, `foobarbaz: https://exmaple.com/open-world-assumption`)`;

IMO its always better to throw an error than silently "fix things" for developers... explicit over implicit is almost always better.

throwing an error teaches the developer how to do things correctly...